src/app/components/catalog/catalog.component.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, OnInit, ViewChild, Input } from '@angular/core';
+import { MatPaginator, MatSort, MatTableDataSource } from '@angular/material';
+import { HttpClient } from '@angular/common/http';
+
+import { merge, Observable, of as observableOf } from 'rxjs';
+import { catchError, map, startWith, switchMap } from 'rxjs/operators';
+import { Product } from '../../models/product';
+import { CatalogService } from '../../services';
+import { CatalogSearch } from '../../models/ProductSearch';
+import { Category, CategorySearch } from '../../models/category';
+import { CaruselComponent } from './carusel/carusel.component';
+import { ProductProperties } from '../../models/product-properties';
+import { ProductPrice } from '../../models/product-price';
+
+@Component({
+  selector: 'app-catalog',
+  templateUrl: './catalog.component.html',
+  styleUrls: ['./catalog.component.css']
+})
+export class CatalogComponent implements OnInit {
+  products: Product[];
+  categories: Category[];
+  displayedColumns: string[] = ['image', 'name', 'price',];
+  dataSource: MatTableDataSource<Product>;
+  filterByCategory: string;
+
+  resultsLength = 0;
+  isLoadingResults = true;
+  isRateLimitReached = false;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+
+  test: any;
+
+  constructor(private http: HttpClient, private catalogService: CatalogService) {
+
+  }
+
+  ngOnInit() {
+    this.catalogService.getAllCategories()
+      .subscribe(
+        (data: CategorySearch) => {
+          this.categories = data.categories;
+        },
+        error => console.log(error)
+      );
+
+    this.catalogService.getAllProducts()
```

Comment:
catalogService should return products object with application domain  types instead of API dtos

Reply:
Created task on the board
https://github.com/VirtoCommerce/vc-procurement-portal-theme/issues/70